cmd/cem.go

```diff
@@ -0,0 +1,395 @@
+package cmd
+
+import (
+	"encoding/gob"
+	"fmt"
+	"log"
+	"os"
+	"syscall"
+
+	"github.com/infamousjoeg/cybr-cli/pkg/cybr/helpers/prettyprint"
+	"github.com/quincycheng/cem-api-go/pkg/cem"
+	"github.com/spf13/cobra"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	// CemOrganization Organization tenant name
+	CemOrganization string
+
+	// CemNonInteractive NonInteractive logon
+	CemNonInteractive bool
+
+	// CemPlatform Platform Name
+	CemPlatform string
+	// CemAccountID Account ID
+	CemAccountID string
+	// CemEntityID Entity ID
+	CemEntityID string
+
+	// CemNonFullAdmin non-full Admin only
+	CemNonFullAdmin bool
+	// CemNonShadowAdmin non-sadow Admin only
+	CemNonShadowAdmin bool
+	// CemFullAdmin Full Admin only
+	CemFullAdmin bool
+	// CemShadowAdmin Shadow Admin
+	CemShadowAdmin bool
+
+	// CemNextToken Next Token
+	CemNextToken string
+
+	// CemSessionTokenPath path to session token file
+	CemSessionTokenPath string = "/.cybr/cem.config"
+
+	// CemEnvAPIKey environment variable of CEM API Key for non-interfactive logon
+	CemEnvAPIKey string = "CEM_APIKEY"
+)
+
+var cemCmd = &cobra.Command{
+	Use:   "cem",
+	Short: "CEM actions",
+	Long: `All actions that can be performed with the Cloud Entitlements Manager.
+	
+	Example Usage:
+	Get remediations on an entity: 
+	$ cybr cem get-remediations -p PLATFORM -a ACCOUNT_ID -e ENTITY_ID`,
+}
+
+var cemLoginCmd = &cobra.Command{
+	Use:   "login",
+	Short: "Login to CEM REST API",
+	Long: `Authenticate to Cloud Entitlements Manager REST API.
+	
+	Example Usage:
+	$ cybr cem login -a $ORGANIZATION
+	
+	For non-interactive logon:
+	$ export ` + CemEnvAPIKey + `=<Your CEM access key>
+	$ cybr cem login -a $ORGANIZATION --non-interactive`,
+	Run: func(cmd *cobra.Command, args []string) {
+		apikey := os.Getenv(CemEnvAPIKey)
+
+		if !CemNonInteractive {
+			// Get secret value from STDIN
+			fmt.Print("Enter API Key: ")
+			byteSecretVal, err := terminal.ReadPassword(int(syscall.Stdin))
+			fmt.Println()
+			if err != nil {
+				log.Fatalln("An error occurred trying to read API key from " +
+					"Stdin. Exiting...")
+			}
+			apikey = string(byteSecretVal)
+		}
+
+		if apikey == "" {
+			log.Fatalf("Provided API Key is empty")
+		}
+
+		token, err := cem.Login(CemOrganization, apikey)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = SaveToken(token)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Printf("Successfully logged onto CEM (organization: %s)\n", CemOrganization)
+	},
+}
+
+// getUserHomeDir Get the Home directory of the current user
+func getUserHomeDir() (string, error) {
```

Comment:
This is currently implemented here. https://github.com/infamousjoeg/cybr-cli/blob/d0951cd0d16707b7e126eae174fd46e444fdd516/pkg/cybr/api/client.go#L20

We should definitely move it to a helper function that is outside of the `cybr` package since it is not related.